modules/admin-ui/src/main/webapp/scripts/shared/directives/segmentsDirective.js

```diff
@@ -181,6 +181,11 @@ angular.module('adminNg.directives')
              * Make sure that times are updates if needed.
              */
         scope.$root.$on('segmentTimesUpdated', function () {
+          angular.forEach(scope.video.segments, function (segment) {
+            if (segment.end > scope.video.duration) {
+              segment.end = scope.video.duration;
+            }
+          });
```

Comment:
Could this be added to the loop before it? Is there any issue that "segment.endTime" will still be old time that is greater than scope.video.duration? In other words, should the "if" check happen before segment.endTime is assigned?

Reply:
I don't understand exactly what you mean with "the loop before it", but the "if" check has to happen before the assignment of `segment.endTime`. I can put the "if" statement into the loop of `scope.setHumanReadableTimes()` (maybe that's what you mean). Then it wouldn't iterate 2 times over the segments.

Reply:
HI @marwyg , upon revisiting this pull, I don't see which loop I was thinking off. So I'm closing this comment!